dev/scripts/build.ts

```diff
@@ -3,46 +3,139 @@ import { join } from "node:path"
-    const outDir = join(outRoot, module === "commonjs" ? "cjs" : "mjs")
-    writeJson(tempTsConfig, {
-        ...tsConfigData,
-        include: ["src"],
+const nuke = (target: string) =>
+    rmSync(target, { recursive: true, force: true })
+const clone = (from: string, to: string): void =>
+    cpSync(from, to, { recursive: true, force: true })
+
+const writeManifest =
+    (overrides: Record<string, unknown>) =>
+    (sourceDir: string, targetDir: string) => {
+        const manifest = readJson(join(sourceDir, "package.json"))
+        writeJson(join(targetDir, "package.json"), {
+            ...manifest,
+            ...overrides
+        })
+    }
+
+const Sources = {
+    utils: ["dev", "utils"],
+    attest: ["dev", "attest"]
+} as const
+
+const replacementDictionary: ReplacementDictionary = {
+    attest: {
+        pattern: /"(\.\.\/)+[^"]*attest\/[^"]*\.js"/g,
+        replacement: `"@arktype/attest"`
+    },
+    utils: {
+        pattern: /"(\.\.\/)+[^"]*utils\/[^"]*\.js"/g,
+        replacement: `"@arktype/utils"`
+    }
+} as const
+
+const ignorePaths = [
+    "node_modules",
+    "package.json",
+    `${tempTsConfigBaseName}.tsbuildinfo`
+]
+
+const fixBuildPaths: (buildPath: string) => void = rewritePaths(
+    replacementDictionary,
+    ignorePaths
+)
+
+const buildFormat = (module: ModuleKind) => {
+    const moduleKindDir = ModuleKindToDir[module]
+    const outDir = join(outRoot, moduleKindDir)
+    const utilsSrc = join(outDir, ...Sources.utils, "src")
+    const attestSrc = join(outDir, ...Sources.attest, "src")
+    const utilsTarget = join(
+        packageRoot,
+        ...Sources.utils,
+        "dist",
+        moduleKindDir
+    )
+    const attestTarget = join(
+        packageRoot,
+        ...Sources.attest,
+        "dist",
+        moduleKindDir
+    )
+
+    const tempTsConfig = {
+        ...baseTsConfig,
+        include: ["src", Sources.utils.join("/"), Sources.attest.join("/")],
         compilerOptions: {
-            ...tsConfigData.compilerOptions,
+            ...compilerOptions,
             noEmit: false,
             module,
             outDir
         }
+    }
+
+    const writePackageManifest = writeManifest({
+        type: ModuleKindToPackageType[module]
     })
+
+    writeJson(tempTsConfigPath, tempTsConfig)
+
     try {
-        shell(`pnpm tsc --project ${tempTsConfig}`)
+        shell(`pnpm tsc --project ${tempTsConfigPath}`)
         const outSrc = join(outDir, "src")
+        const outDev = join(outDir, "dev")
         // not sure which setting to change to get it to compile here in the first place
-        cpSync(outSrc, outDir, {
-            recursive: true,
-            force: true
-        })
-        if (module === "commonjs") {
-            writeJson(join(outDir, "package.json"), {
-                type: "commonjs"
-            })
-        }
-        rmSync(outSrc, { recursive: true, force: true })
+        clone(outSrc, outDir)
```

Comment:
Is it possible to refactor this so we call a single function that handles these steps for each package? Not a big deal if not possible, just would make it easier to extend that way.

Reply:
I tried refactoring this out, but due to small differences between each of them, it ended up being more trouble than it's worth.

If we add another package I think there will be enough duplication between that one and `attest` that we could write a function for those 2 that any others will be able to use going foward